src/middlewares/generator-template-parameters-validation.middleware.ts

```diff
@@ -0,0 +1,87 @@
+import fs from 'fs';
+import path from 'path';
+
+import Ajv from 'ajv';
+import { Request, Response, NextFunction } from 'express';
+
+import { ProblemException } from '../exceptions/problem.exception';
+
+/**
+ * Validate for the given template the template's parameters.
+ */
+export async function generatorTemplateParametersValidationMiddleware(req: Request, _: Response, next: NextFunction) {
```

Comment:
Question: Shouldn't be this just part of the controller instead of a separate middleware? Middlewares are not meant to be reused most of them? I feel confused.

Reply:
Each middleware that is used in codebase I wanted to write in a separate folder. Do you think to change this middleware and tie its functionality to the `/generate` controller itself?

Reply:
Ok, moved logic to the controller :)